src/backend/tests/unmocked/onboarding.test.ts

```diff
@@ -502,4 +502,114 @@ describe('Onboarding tests', () => {
       expect(newChildChecklist!.dateDeleted).not.toBeNull();
     });
   });
+
+  describe('Toggle Checklist Item', () => {
+    it('Toggles checklist item and updates the parent checklist item if all children checked', async () => {
```

Comment:
This is a great thourough check. Can you maje it so that you are testing for every single exception that can be thrown as well. You will need multiple `it` statements. it might also be easier to seperate these tests you have in different it blocks. for example you woul dhave to to check an item, one to uncheck an item. one to check the parent checklist once you check the last subtask and so on like that